### ISSUE TITLE
CMake build/install dependency as a part of fips build?

### ISSUE THREAD

**Issue by russpowers** [floooh/fips#179]
Hi, I'm trying to write a fips wrapper for libwebsockets, but it's got a fairly involved CMake setup, and I'd rather not try to pull everything out into a custom fips CMakeFile.txt.  I would like to just have fips use CMake to build/install libwebsockets (install it locally, like into a subdirectory of fips-libwebsockets), and then I could use the output of that in other fips projects.  Is that possible?  Or am I approaching this the wrong way?  Any advice would be greatly appreciated!

**Comment by russpowers** [comment]
Thanks, that worked!  Is there any way to get fips to add a subdirectory of the installed library as an include directory?  libwebsockets generates a lws_config.h that needs to be included for things to work.  For now, I just copied the generated one to include/libwebsockets: https://github.com/russpowers/fips-libwebsockets.

**Comment by floooh** [comment]
You can add a ```target_include_directories``` statement to your fips-wrapper CMakeLists.txt file with the directory where the config.h file is generated, something like this:

https://github.com/fips-libs/fips-soloud/blob/4cb8def50156a08e369d190f23acc23fbd297507/CMakeLists.txt#L119

You can experiment with the PUBLIC arg, if the header file is only included by libwebsockets implementation files, a PRIVATE would be better (see here: https://cmake.org/cmake/help/v3.0/command/target_include_directories.html).

**Comment by russpowers** [comment]
I wasn't able to get `target_include_directories` working, CMake was complaining that I couldn't add them to another project (or, I suppose, a subproject?).  I did figure something out though, I added a `fips-include.cmake`: https://github.com/russpowers/fips-libwebsockets/blob/master/fips-include.cmake.  It seems to be working well, although I've only tested it on Windows builds.  

Thanks again for your help!
